VMware.CloudFoundation.CertificateManagement.psm1

```diff
@@ -1319,6 +1319,415 @@ Function Install-EsxiCertificate {
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $vcfCertCaUsername,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $vcfCertCaPassword,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $vcfCertCaTemplate
+    )
+
+    # Connect to VMware SDDC Manager
+    if (Test-VCFConnection -server $server) {
+        $connectionStatus = Test-NetConnection -Port 443 -ComputerName $vcfCertCaFqdn
+        if ($connectionStatus.TcpTestSucceeded -eq "True" ) {
+            if (Test-VCFAuthentication -server $server -user $user -pass $pass) {
+                $vcfVersion = Get-VCFManager | select version | Select-String -Pattern '\d+\.\d+' -AllMatches | ForEach-Object {$_.matches.groups[0].value}
+                $serverUrl = "https://$vcfCertCaFqdn/certsrv"
+                Try
+                {
+                    Write-Output "Starting the Process of Configuring a Microsoft Certificate Authority in SDDC Manager"
+                    Write-Output "Checking Microsoft Certificate Authority Configuration"
+                    $vcfCertCa = Get-VCFCertificateAuthority
+                    If ($vcfCertCa.username -ne "$vcfCertCaUsername")
+                    {
+                        Write-Output "Configuring Microsoft Certificate Authority Connection in SDDC Manager using ($vcfCertCaUsername)"
+                        Set-VCFMicrosoftCA -serverUrl $serverUrl -username $vcfCertCaUsername -password $vcfCertCaPassword -templateName $vcfCertCaTemplate | Out-Null
+                        Write-Output "Configuration of Microsoft Certificate Authority in SDDC Manager Using ($($vcfCertCaUsername)): SUCCESSFUL"
+                    } else {
+                        Write-Output "Configuration of Microsoft Certificate Authority in SDDC Manager Using ($($vcfCertCa.username)), already exists: SKIPPED"
+                    }
+                    Write-Output "Finished the Process of Configuring a Microsoft Certificate Authority in SDDC Manager"
+                } 
+                Catch 
+                {
+                    $ErrorMessage = $_.Exception.Message
+                    Write-Output "Error was: $ErrorMessage"
+                }
+            } else {
+                Write-Error "Unable to Authenticate SDDC Manager ($server)"
+            }
+        } else {
+            Write-Error "Unable to connect to Microsoft Certificate Authority ($vcfCertCaFqdn)"
+        }
+    } else {
+        Write-Error "Unable to connect to SDDC Manager ($server)"
+    }
+}
+
+Function gatherSddcInventory {
+    Param (
+        [Parameter (Mandatory = $true)] $domainType,
+        [Parameter (Mandatory = $true)] $workloadDomain
+    )
+
+    # Gathers various deployment details from SDDC Manager
+    $sddcMgr = Get-VCFManager
+    $sddcMgrVersion = $sddcMgr.version.split(".")[0]
+
+    $resourcesObject = @()
+    # SDDC Manager
+    IF ($domainType -eq "Management")
+    {
```

Comment:
```suggestion
    If ($domainType -eq "Management") {
```

Reply:
addressed in (131acd4bec5c68a803eb55ef8c6b8d2638b81d17)